### PR TITLE
fix(CX-3780): show artwork image inside my collection when gemini processing fails

### PR DESCRIPTION
**Pull request by MounirDhahri**
The type of this PR is: **fix**

This PR solves [CX-3780]

### Description
We are currently saving the large version of the image as image url assuming the latter always exists. This is not the case though when Gemini fails. We are updating the logic now to parse the available image versions and use the first biggest image available.

**Next step:** Introduce the same fix to **Eigen**

| Before                                                                              | After                                                                               |
|-------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------|
| https://github.com/artsy/force/assets/11945712/4941487c-c6b6-4883-8ca8-aa391f52f7c2 | https://github.com/artsy/force/assets/11945712/3fa28e9c-5fd6-45a8-bf62-a51857076eca |

Thanks for reporting this @pvinis 

[GRO-434]: https://artsyproduct.atlassian.net/browse/GRO-434?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CX-3780]: https://artsyproduct.atlassian.net/browse/CX-3780?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by dariakoko — APPROVED** [review]

File: src/Apps/MyCollection/Routes/EditArtwork/Components/MyCollectionArtworkFormArtworkStep.tsx [hunk 1]
```diff
@@ -57,12 +57,29 @@ export const MyCollectionArtworkFormArtworkStep: React.FC<MyCollectionArtworkFor
...
+
+      const imageVersion = imageVersionsSortedBySize.find(version =>
+        image?.imageVersions?.includes(version)
+      )
+
+      return {
+        name: "Automatically added",
+        url: !!imageVersion
+          ? image?.imageURL?.replace(":version", imageVersion)
+          : null,
```
Comment (olerichter00) [inline src/Apps/MyCollection/Routes/EditArtwork/Components/MyCollectionArtworkFormArtworkStep.tsx:78]: Could we pass the list of versions directly in the GrqphQL quer instead ([example](https://github.com/artsy/force/blob/b64d9edf38df65a3a2733799771cd2f09c131e4d/src/Apps/Article/Components/Sections/ArticleSectionImageCollection/ArticleSectionImageCollectionImage.tsx#L66))? I think this case should be supported 🤔

Comment (MounirDhahri) [reply src/Apps/MyCollection/Routes/EditArtwork/Components/MyCollectionArtworkFormArtworkStep.tsx:78]: didn't know we can do that! does this pattern also return the first one available in the provided array?

Comment (olerichter00) [reply src/Apps/MyCollection/Routes/EditArtwork/Components/MyCollectionArtworkFormArtworkStep.tsx:78]: Yes, I think it does exactly what we need. It goes through `imageVersions` and looks for the first version that exists ([code](https://github.com/artsy/metaphysics/blob/eb9a1362323222a5bd22089c4b798f1442af4a89/src/schema/v2/image/normalize.ts#L24)).

You should be able to get the image URL like this:

```graphql
image {
  url(
    version: [
        "main",
        "normalized",
        "larger",
        "large",
        "medium",
        "small"
    ]
  )
}
```

Comment (MounirDhahri) [reply src/Apps/MyCollection/Routes/EditArtwork/Components/MyCollectionArtworkFormArtworkStep.tsx:78]: great, thanks @olerichter00. I will update

Comment (MounirDhahri) [reply src/Apps/MyCollection/Routes/EditArtwork/Components/MyCollectionArtworkFormArtworkStep.tsx:78]: it seems like the version string is not supported within the Image interface in MP. I don't believe it makes sense to add support to it in this context, what do you think?

Comment (olerichter00) [reply src/Apps/MyCollection/Routes/EditArtwork/Components/MyCollectionArtworkFormArtworkStep.tsx:78]: I updated the query in `MyCollectionArworkSearch.tsx` and it appears to be working correctly.
```
images {
                  height
                  internalID
                  isDefault
                  url(version: ["large", "tall"])
                  width
                }
```

Comment (MounirDhahri) [reply src/Apps/MyCollection/Routes/EditArtwork/Components/MyCollectionArtworkFormArtworkStep.tsx:78]: I used `imageUrl` initially , i can try this out as well 👀

Comment (MounirDhahri) [reply src/Apps/MyCollection/Routes/EditArtwork/Components/MyCollectionArtworkFormArtworkStep.tsx:78]: worked! thanks

**Pull request merged**